extensions/oplogPopulator/OplogPopulator.js

```diff
@@ -198,27 +202,57 @@ class OplogPopulator {
     }
 
     /**
-     * start listening to bucket
+     * assigns a connector to a bucket
+     * @param {string} bucketName bucket name
+     * @returns {string} assigned connector for bucket
+     */
+     _assignConnectorForBucket(bucketName) {
+        const assignedConnector = this._connectorToBucket.get(bucketName);
+        // return previously assigned bucket if it exists
+        if (assignedConnector) {
+            return assignedConnector;
+        } else {
+            // else return the connector with the fewest buckets assinged to it
+            const bucketPerConnector = this._configuredConnectors.map(connector =>
+                this._bucketsForConnectors[connector].size);
+            const min = Math.min(...bucketPerConnector);
+            const idx = bucketPerConnector.indexOf(min);
+            return this._configuredConnectors[idx];
+        }
+    }
+
+    /**
+     * start listening to a specific bucket's events
      * @param {string} bucketName bucket name
+     * @param {boolean} updateConnector should connector get updated
      * @returns {undefined}
      */
-    _listenToBucket(bucketName) {
-        this._bucketsForConnector.add(bucketName);
-        this.updateConnectorPipeline(constants.defaultConnectorName, [...this._bucketsForConnector]);
+    _listenToBucket(bucketName, updateConnector = true) {
+        const assignedConnector = this._assignConnectorForBucket(bucketName);
+        this._connectorToBucket.set(bucketName, assignedConnector);
```

Comment:
the scope/role of the assignConnectorForBucket, and the followup "assignation", is not very consistent : either it should do indeed the assignation (and update the data), only simply handle the "retrieval" (and we must assign here if not already done)